pkg/nfd-master/nfd-master.go

```diff
@@ -71,6 +71,16 @@ type ExtendedResources map[string]string
 // Annotations are used for NFD-related node metadata
 type Annotations map[string]string
 
+// Restrictions contains the restrictions on the NF and NFR Crs
+type Restrictions struct {
+	NodeFeatureNamespaceSelector *metav1.LabelSelector
+	MaxLabelsPerCR               int
+	MaxTaintsPerCR               int
+	MaxExtendedResourcesPerCR    int
+	DenyNodeFeatureLabels        bool
+	OverwriteLabels              bool
```

Comment:
Suggesting a slight rename and change in functionality. If we add this, I think we should treat labels, annotations, taints and ERs the same way. I.e. allow overwriting previous values or then deny.

```suggestion
	AllowOverwrite              bool
```

WDYT?